design/workitems.go

```diff
@@ -150,4 +150,19 @@ var _ = a.Resource("workitem", func() {
 		a.Response(d.NotFound, JSONAPIErrors)
 		a.Response(d.Unauthorized, JSONAPIErrors)
 	})
+	a.Action("reorder", func() {
+		a.Security("jwt")
+		a.Routing(
+			a.PATCH("/reorder"),
+		)
+		a.Description("reorder the work item with the given id.")
+		a.Payload(workItemSingle)
```

Comment:
The payload can not be just a single workitem.

If you look at the UI design and behavior of the WorkItemList reorder feature, it is suppose to support: Move X, Y, .. Before Z